app/main.py

```
@@ -1,25 +1,72 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, owner: Any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value should be an integer")
+        if not self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, "access denied")
```

If there is an incorrect value, you should raise an error without setting the attribute